code/modules/power/treadmill.dm

```diff
@@ -51,19 +51,17 @@
 /obj/machinery/power/treadmill/proc/powerwalk(atom/movable/AM as mob)
 	if(!ismob(AM)) return //Can't walk on a treadmill if you aren't animated
 	if(get_turf(AM) != loc) return //Can't bump from the outside
-	var/mob/runner = AM
-	if(!istype(runner,/mob/living/simple_animal)&&runner.bodytemperature <= 360)
-		runner.bodytemperature += 2 //Same heating pattern as being fat
-	if(runner.nutrition && runner.stat != DEAD)
-		runner.nutrition -= HUNGER_FACTOR*2 //Running on a treadmill makes you hungry fast
-	flick("treadmill-running", src)
-	playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
-	var/calc = DEFAULT_BUMP_ENERGY * power_efficiency * runner.treadmill_speed
-	if(runner.reagents) //Sanity
-		for(var/datum/reagent/R in runner.reagents.reagent_list)
-			calc *= R.sport
-	if(M_HULK in runner.mutations) calc *= 5
-	count_power += calc
+	var/mob/living/runner = AM
+	if((runner.bodytemperature < T0C + 52) && runner.burn_calories(HUNGER_FACTOR/2))
+		flick("treadmill-running", src)
+		playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
+		var/calc = DEFAULT_BUMP_ENERGY * power_efficiency * runner.treadmill_speed
+		if(runner.reagents) //Sanity
+			for(var/datum/reagent/R in runner.reagents.reagent_list)
+				calc *= R.sport
+		if(M_HULK in runner.mutations) calc *= 5
+		count_power += calc
+	else to_chat(runner,"<span class='warning'>You're exhausted! You can't run anymore!</span>")
```

Comment:
It stops them running if they:
a) Can't burn enough calories (duh, that makes sense)
b) Are >52C, which takes a fair while to get to now.